### PR TITLE
Fixing errors caused by new core

### PR DESCRIPTION
**Pull request by dusan01**
## Description

There was an issue when you started a resource about core objects, so this fixes it, I also moved the config to a folder and adjusted the code a little bit to match the new format and more user-friendly names.

There is still an error for the camera.lua file, I was unable to determine what's causing it, so if someone can point me in the right direction, I will be more than happy to fix that issue too.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by alberttheprince** [comment]
Tested these fixes on live server, it seems to fix the job and make it work with Qbox. However, the resource itself seems lackluster/lacking in design. Maybe potential for reworking slightly?

File: client/camera.lua [hunk 1]
```diff
@@ -61,8 +61,7 @@ local function CheckInputRotation(cam, zoomvalue)
 end
 
 local function HandleZoom(cam)
-	local lPed = PlayerPedId()
-	if not ( IsPedSittingInAnyVehicle( lPed ) ) then
+	if not ( IsPedSittingInAnyVehicle(cache.ped) ) then
```
Comment (Manason) [inline client/camera.lua]:
```suggestion
	if not cache.vehicle then
```

File: client/camera.lua [hunk 2]
```diff
@@ -102,43 +101,32 @@ local function Breaking(text)
...
-	AddTextComponentString(string)
-    DisplayHelpTextFromStringLabel(0, 0, 1, -1)
-end
-
 ---------------------------------------------------------------------------
 -- Toggling Cam --
 ---------------------------------------------------------------------------
 
-RegisterNetEvent("Cam:ToggleCam", function()
+RegisterNetEvent('qbx_newsjob:client:toggleCam', function()
```
Comment (Manason) [inline client/camera.lua:112]: This seems like it should be a callback rather than an event

File: client/camera.lua [hunk 3]
```diff
@@ -148,24 +136,20 @@ end)
...
-				DisableControlAction(0,25,true) -- disable aim
-				DisableControlAction(0, 44,  true) -- INPUT_COVER
-				DisableControlAction(0,37,true) -- INPUT_SELECT_WEAPON
-				SetCurrentPedWeapon(PlayerPedId(), GetHashKey("WEAPON_UNARMED"), true)
-				Wait(7)
+				DisablePlayerFiring(cache.playerId, true)
+				DisableControlAction(0,25, true)
+				DisableControlAction(0, 44, true)
+				DisableControlAction(0,37, true)
+				SetCurrentPedWeapon(cache.ped, joaat(`WEAPON_UNARMED`), true)
```
Comment (Manason) [inline client/camera.lua]:
```suggestion
				SetCurrentPedWeapon(cache.ped, `WEAPON_UNARMED`, true)
```
backticks already convert a string to a hash

File: client/camera.lua [hunk 4]
```diff
@@ -181,35 +165,34 @@ end)
...
+					AttachCamToEntity(cam1, cache.ped, 0.0,0.0,1.0, true)
+					SetCamRot(cam1, 2.0,1.0,GetEntityHeading(cache.ped))
 					SetCamFov(cam1, fov)
 					RenderScriptCams(true, false, 0, 1, 0)
-					PushScaleformMovieFunction(scaleform, "security_camera")
+					PushScaleformMovieFunction(scaleform, 'security_camera')
 					PopScaleformMovieFunctionVoid()
 
-					while movcamera and not IsEntityDead(lPed) and (GetVehiclePedIsIn(lPed) == vehicle) and true do
+					while movcamera and not IsEntityDead(cache.ped) and (GetVehiclePedIsIn(cache.ped) == vehicle) and true do
```
Comment (Manason) [inline client/camera.lua]:
```suggestion
					while movcamera and not IsEntityDead(cache.ped) and cache.vehicle == vehicle) do
```

File: client/camera.lua [hunk 5]
```diff
@@ -181,35 +165,34 @@ end)
...
-					local scaleform = RequestScaleformMovie("security_camera")
+					local scaleform = lib.requestScaleformMovie('security_camera')
 					while not HasScaleformMovieLoaded(scaleform) do
 						Wait(10)
 					end
 
-					local lPed = PlayerPedId()
-					local vehicle = GetVehiclePedIsIn(lPed)
-					local cam1 = CreateCam("DEFAULT_SCRIPTED_FLY_CAMERA", true)
+					local vehicle = GetVehiclePedIsIn(cache.ped)
```
Comment (Manason) [inline client/camera.lua]:
```suggestion
					local vehicle = cache.vehicle
```

File: client/camera.lua [hunk 6]
```diff
@@ -338,26 +315,22 @@ end)
 --B Mic --
 ---------------------------------------------------------------------------
 
-RegisterNetEvent("Mic:ToggleBMic", function()
+RegisterNetEvent('qbx_newsjob:client:toggleBMic', function()
```
Comment (Manason) [inline client/camera.lua:318]: Seems like it should be a callback

File: client/main.lua [hunk 7]
```diff
@@ -1,104 +1,110 @@
...
+        local netId = lib.callback.await('qbx_newsjob:server:spawnHeli', false, vehicleInfo, coords, Lang:t('info.news_plate')..tostring(math.random(1000, 9999)), true)
+        local timeout = 100
+        while not NetworkDoesEntityExistWithNetworkId(netId) and timeout > 0 do
+            Wait(10)
+            timeout = timeout - 1
+        end
+        local veh = NetToVeh(netId)
+        if veh == 0 then
+            exports.qbx_core:Notify(Lang:t('error.cant_spawn_vehicle'), 'error')
+            return
```
Comment (Manason) [inline client/main.lua]: How much of this is duplicate code with spawning a vehicle? Consider extracting a function

File: client/main.lua [hunk 8]
```diff
@@ -311,24 +307,116 @@ CreateThread(function()
...
+CreateThread(function()
+    while true do
+        Wait(0)
+        local inRange = false
+
+        if IsLoggedIn then
+            local pos = GetEntityCoords(cache.ped)
+            if #(pos - vector3(config.locations.helicopterStorageEntrance.coords.xyz)) < 1.5 or #(pos - vector3(config.locations.helicopterStorageExit.coords.xyz)) < 1.5 then
+                inRange = true
+                if #(pos - vec3(config.locations.helicopterStorageEntrance.coords.xyz)) < 1.5 then
```
Comment (Manason) [inline client/main.lua]: Why are these distance checks needed? Seems like these zones should be created once; and not created and destroyed within a loop. Could clean them up on resource stop and create them on resource start though.

File: server/main.lua [hunk 9]
```diff
@@ -1,22 +1,45 @@
...
-QBCore.Commands.Add("newsmic", "Grab a news microphone", {}, false, function(source, _)
-    local Player = QBCore.Functions.GetPlayer(source)
-    if Player.PlayerData.job.name == "reporter" then
-        TriggerClientEvent("Mic:ToggleMic", source)
-    end
+lib.addCommand('newsbmic', {
+    help = Lang:t('info.newsbmic'),
+}, function(source)
+    local Player = exports.qbx_core:GetPlayer(source)
+    if Player.PlayerData.job.name ~= 'reporter' then return end
```
Comment (Manason) [inline server/main.lua]: consider extracting duplicate code into a function "isAReporter"

**Review by Manason — CHANGES_REQUESTED** [review]

**Comment by dusan01** [comment]
I do agree with both of you, this does actually need a good rewrite, but with this PR I just wanted to convert it and make it usable, before doing the actual rewrite to make it better.

File: client/main.lua (hunk 8 above)
Comment (Manason) [reply client/main.lua]: The other comments are fine to leave for another PR, but this one has to do with code introduced here and should be addressed.

**Review by Manason — APPROVED** [review]

**Review by mafewtm — APPROVED** [review]

**Pull request merged**